### DISCUSSION TITLE
Dealing with connection refused errors with high number of concurrent requests?

### DISCUSSION THREAD

**Discussion by lewisl9029** [seaweedfs/seaweedfs#3437] [Q&A] [answered]
I have a use case where sometimes hundreds of files need to be loaded concurrently from seaweed, and I was seeing more and more errors pop up in production as I start getting more traffic in different regions.

When testing locally, it seems like somewhere between 100-200 concurrent requests is where seaweed will start refusing connections. Not entirely sure if this is something that can be tweaked in a setting somewhere or if it's a fundamental limit?

Currently I've been limiting connections on the client side using connection pools, and it's working fine so far, but I fear that might not scale well once I begin to increase the number of clients. Curious if anyone has any suggestions on other ways to deal with this connection limit?

Any thoughts would be appreciated!

**Comment by chrislusf** [comment, marked as answer]
* Front the filer with a reverse proxy
* Add more filers
* Connection pool

**Reply by lewisl9029** [reply]
Thanks for the quick suggestions! Fronting with a reverse proxy is an option I've been looking into more, since I've already been using connection pools and adding more filers would add more complexity around keeping them in sync. 

Nginx looks like it isn't able to queue once max connection is reached in the free version according to https://stackoverflow.com/questions/37879337/nginx-limit-the-number-of-connections-to-an-upstream-server

Though I wonder if maybe just configuring keepalive options will reduce max connections enough for this to be no longer a problem for the most part: https://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive

Curious if you have any more thoughts on what reverse proxy configurations will help most here?

**Reply by chrislusf** [reply]
haproxy

**Reply by lewisl9029** [reply]
Ah awesome, looks like HAProxy supports connection queuing out of the box: https://www.haproxy.com/blog/four-examples-of-haproxy-rate-limiting/

Will give it a shot. Thank you!

**Comment by LuckyFrost** [comment]
What do you mean by "refusing connections"?
